Source/AliveLibAE/Input.cpp

```diff
@@ -1379,38 +1400,56 @@ EXPORT int Input_Convert_KeyboardGamePadInput_To_Internal_Format_492150()
                             {
                                 pressed_keyboard_keys = keys_down;
                                 input_command_delete_pressed = 1;
+                                ++shoulderButtonsPressedCount;
                             }
                         }
-                    }
 
-                    if (input_command_c_pressed)
-                    {
-                        if (input_command_delete_pressed)
-                        {
-                            pressed_keyboard_keys |= InputCommands::Enum::eChant;
-                        }
-                        else
+                        if (sGamePadBindings_5C98E0[i] & InputCommands::Enum::eRun)
                         {
-                            if (pButtons & InputCommands::Enum::eUp)
-                            {
-                                pressed_keyboard_keys |= InputCommands::Enum::eGameSpeak2;
-                            }
-                            if (pButtons & InputCommands::Enum::eDown)
-                            {
-                                pressed_keyboard_keys |= InputCommands::Enum::eGameSpeak3;
-                            }
-                            if (pButtons & InputCommands::Enum::eLeft)
+                            if ((1 << i) & pButtons)
```

Comment:
can we not explicitly check the required button bits instead of doing it this way? this way is harder to understand what is happening

Reply:
Just to clarify - are you asking me to modify the way the game checks for eSpeak1 and eSpeak2?
Note: I haven't modified this - I've just added a check for eRun and eSneak in the same loop.

Reply:
I'm talking about ` if ((1 << i) & pButtons)` which seems to be a new addition?

Reply:
That was used for eSpeak1 and eSpeak2 as well. I've just followed its implementation.